### PR TITLE
Make Container block a functional component

### PR DESCRIPTION
**Pull request by JeanPaiva**
This PR makes the container block a functional React component

### CODE REVIEW

File: src/blocks/container/components/InspectorControls.js [hunk 1]
```diff
@@ -0,0 +1,2136 @@
...
+		removeVerticalGap,
+		removeVerticalGapTablet,
+		removeVerticalGapMobile,
+		orderTablet,
+		orderMobile,
+		align,
+		shapeDividers,
+	} = attributes;
+
+	const hasGridContainer = false,
```
Comment (tomusborne) [inline src/blocks/container/components/InspectorControls.js:96]: Missing the logic to define this.

File: src/hoc/withContainerLegacyMigration.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		const {
+			clientId,
+			attributes,
+			setAttributes,
+		} = props;
+
+		useEffect( () => {
+			const thisBlock = document.getElementById( `block-${ clientId }` );
+
+			if ( thisBlock && 'full' === attributes.align ) {
```
Comment (tomusborne) [inline src/hoc/withContainerLegacyMigration.js:17]: This should likely go into `edit`.

File: src/blocks/container/edit.js [hunk 3]
```diff
@@ -195,2406 +143,91 @@ class GenerateBlockContainer extends Component {
...
-		if ( ! getPreviewDeviceType ) {
-			return {
-				media: featuredImageId ? getMedia( featuredImageId ) : null,
-				deviceType: null,
-				hasChildBlocks: blockEditor ? 0 < blockEditor.getBlockOrder( clientId ).length : false,
-			};
-		}
-
 		return {
 			media: featuredImageId ? getMedia( featuredImageId ) : null,
```
Comment (tomusborne) [inline src/blocks/container/edit.js:227]: I don't actually think we're using this in the block anywhere. I believe it was leftover from when I tried to build a featured image as background image feature.

File: src/blocks/container/edit.js [hunk 4]
```diff
@@ -195,2406 +143,91 @@ class GenerateBlockContainer extends Component {
...
-			deviceType: getPreviewDeviceType(),
 			hasChildBlocks: blockEditor ? 0 < blockEditor.getBlockOrder( clientId ).length : false,
```
Comment (tomusborne) [inline src/blocks/container/edit.js:228]: Can we use `useSelect` in the function instead of adding it here?

File: src/blocks/container/edit.js [hunk 5]
```diff
@@ -1,186 +1,134 @@
...
+
+		Object.keys( shapes ).forEach( ( name ) => {
+			allShapes[ name ] = {
+				label: shapes[ name ].label,
+				icon: shapes[ name ].icon,
+			};
+		} );
+	} );
+
+	useEffect( () => {
```
Comment (tomusborne) [inline src/blocks/container/edit.js:127]: I wonder if this only has to fire when `attributes.align` is changed?

File: src/blocks/container/components/ComponentCSS.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+export default ( props ) => {
+	const { deviceType } = props;
+
+	return (
+		<Fragment>
+
+			<MainCSS { ...props } />
+
+			{ deviceType &&
+			<Fragment>
```
Comment (tomusborne) [inline src/blocks/container/components/ComponentCSS.js:17]: Indent issue - seeing this quite a bit inside `edit` as well I think.

File: src/blocks/container/components/InspectorControls.js [hunk 7]
```diff
@@ -0,0 +1,2158 @@
...
+					initialOpen={ true }
+					icon={ getIcon( 'layout' ) }
+					className={ 'gblocks-panel-label' }
+					id={ 'containerLayout' }
+					state={ state }
+				>
+					{ 'Desktop' === deviceType &&
+					<Fragment>
+						{ hasGridContainer &&
+						<ToggleControl
```
Comment (tomusborne) [inline src/blocks/container/components/InspectorControls.js:181]: Indent - happens quite a bit in this file.

File: src/blocks/container/edit.js (hunk 3 above)
Comment (JeanPaiva) [reply src/blocks/container/edit.js:227]: Removed.

File: src/blocks/container/components/InspectorControls.js (hunk 7 above)
Comment (JeanPaiva) [reply src/blocks/container/components/InspectorControls.js:181]: Ye, my IDE formats in this weird way.

File: src/blocks/container/edit.js (hunk 3 above)
Comment (JeanPaiva) [reply src/blocks/container/edit.js]: Also, for the block count I used the existing state for the block count I created for the grid, so it simplified a bit the logic to get the children to count.

File: src/blocks/container/components/ShapeDividers.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+		<Fragment>
+			{ !! attributes.shapeDividers.length &&
+			<div className="gb-shapes">
+				{
+					attributes.shapeDividers.map( ( location, index ) => {
+						const shapeNumber = index + 1;
+
+						return <Fragment key={ index }>
+							{ 'undefined' !== typeof allShapes[ shapeDividers[ index ].shape ] &&
+							<div
```
Comment (tomusborne) [inline src/blocks/container/components/ShapeDividers.js:18]: Indent

**Review by tomusborne — APPROVED** [review]

**Pull request merged**